BookB-Pset_GeoScienceModelCommon

larswik opened the pull request:
# Pset: Pset_GeoScienceModelCommon

## Applicability: IfcGeoScienceModel

## Description: 

Common properties for geoscience models.

| Property | Datatype | Description |
|----------|----------|-------------|
|ModelPurpose|IfcText|For which Purpose and BIM-Usecase was the model created? E.g. stability assessment, qunatity assessment, tendering etc.|
|CreationDate|IfcDate|Creation date of the model.|
|ReleaseDate|IfcDate|Date of release of the model.|
|UpdateDate|IfcDate|Date of update of the model.|
|VersionNumber|IfcLabel|Version tag of the model regarding its descriptive informations.|
|Purpose|IfcLabel|Purpose for which the model was done. Controlled vocabulary to be consolidated: scientific study, mining, oil reservoir, geotechnics, hydrogeology, simulation etc.|
|Status|IfcLabel|Status of the model. Controlled vocabulary to be consolidated : ended, in progress, in development etc. |
|Author|IfcLabel|Name of the author(s), according to ISO 19115 terminology.<br />>NOTE Maps to PROV:Agent.|
|Organization|IfcLabel|Name of the author(s), according to ISO 19115 terminology.<br />>NOTE Maps to PROV:Agent.|
|AreaOfInterest|IfcIdentifier|Reference to geographical gazeetteer entries (France, Paris basin etc). <br />>NOTE Geonames (https://www.geonames.org/) could be used in case no other is available.|

larswik committed:
BookB-Pset_GeoScienceModelCommon

SergejMuhic commented on IFC4x3/Properties/a/Author_1oQcJRV0j0WhD1ng6TcHzF/Documentation.md:
```diff
@@ -0,0 +1,2 @@
+Name of the author(s), according to ISO 19115 terminology .
```
```suggestion
>NOTE Definition according to ISO 19115: Name of the author(s).
```

Michel-Rives replied:
Proposed change is fine

SergejMuhic commented on IFC4x3/Properties/o/Organization_0Ksewl9eXCOBb5HfyICi72/Documentation.md:
```diff
@@ -0,0 +1,2 @@
+Name of the author(s), according to ISO 19115 terminology.
```
This description seems wrong.

Michel-Rives replied:
Please change the description to "Name of the organizations/authors"

SergejMuhic commented on IFC4x3/Properties/p/Purpose_1gm$OqDpv9KQq3uIG1DEsc/Documentation.md:
```diff
@@ -0,0 +1,2 @@
+Purpose for which the model was done.
+>NOTE Controlled vocabulary to be consolidated: scientific study, mining, oil reservoir, geotechnics, hydrogeology, simulation etc.
```
Enum?

Michel-Rives replied:
Please change description to "Purpose could be, eg: scientific study, mining, oil reservoir, geotechnics, hydrogeology, simulation"
Please rename the property name to "ModelPurpose"

larswik replied:
> Please change description to "Purpose could be, eg: scientific study, mining, oil reservoir, geotechnics, hydrogeology, simulation" Please rename the property name to "ModelPurpose"

There is already another property with the name ModelPurpose with a different meaning ("BIM Usecase").

SergejMuhic commented on IFC4x3/Properties/s/Status_16vVl01en6dv09d8hlKChk/Documentation.md:
```diff
@@ -0,0 +1,2 @@
+Status of the model. 
+>NOTE Controlled vocabulary to be consolidated : ended, in progress, in development etc.
```
Enum?

Michel-Rives replied:
Please change description to "Status could be, eg: ended, in-progress, in-development"
Please rename the property name to "ModelStatus"

larswik committed:
Update IFC4x3/Properties/a/Author_1oQcJRV0j0WhD1ng6TcHzF/Documentation.md

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

larswik commented on IFC4x3/Properties/o/Organization_0Ksewl9eXCOBb5HfyICi72/Documentation.md:
```diff
@@ -0,0 +1,2 @@
+Name of the author(s), according to ISO 19115 terminology.
```
```suggestion
Name of the organizations/authors, according to ISO 19115 terminology.
```

larswik committed:
Update IFC4x3/Properties/o/Organization_0Ksewl9eXCOBb5HfyICi72/Documentation.md

larswik committed:
Changes according to comments by @Michel-Rives

Except changing Purpose to ModelPurpose since ModelPurpose already as another property in the Pset.

larswik commented:
From Geo greoup meeting 2023-08-31:
ModelPurpose => ModelUseCase
Purpose: @mbeaufils will look for a definition.

mbeaufils commented:
Dear @larswik, @j-weil

I agree with @Michel-Rives proposals:
- Delete the current "ModelPurpose",
- Rename the current "Purpose" to "ModelPurpose": Purpose for which the model was done. 
- Rename the current "Status" to "ModelStatus": Status of the model. 
- Authors definition: Name of the author(s).
- Organisation definition: `Name of the organisation(s) involved in producing the model.`

Regarding the codelists:
- ModelPurpose: https://data.geoscience.earth/ncl/ModelPurpose
- ModelStatus: https://data.geoscience.earth/ncl/ModelStatus

larswik commented:
ToDo @larswik: 
- Renaming ModelPurpose to ModelUseCase according to the meeting with the domain team this morning.
    - I propose to update the description: "Specifies the use case for which the model was created. ModelUseCase could be eg: stability assessment, quantity assessment, tendering etc." - similarly to the other proposals by @Michel-Rives.
- Renaming Purpose to ModelPurpose.
    - Not sure whether the codelists are mandatory or just examples. I suggest to reuse the proposal from Michel and add text about the codelist, such as "Purpose could be, eg: scientific study, mining, oil reservoir, geotechnics, hydrogeology, simulation or a URI from the codelist https://data.geoscience.earth/ncl/ModelPurpose"
- Rename Status to ModelStatus and handle the codelist similarly to ModelPurpose above:
    - "Status could be, eg: ended, in-progress, in-development or a URI from the codelist https://data.geoscience.earth/ncl/ModelStatus"
- Regarding Author/Organisation: Should these actually be list values (since it obviously can be plural)? Should we rename to "Authors" and "Organisations"? Should we also make the Authors description analogue to Organisations, e.g. "Name of the author(s) involved in producing the model". Are these two properties unrelated? ping @mbeaufils.

larswik committed:
Merge remote-tracking branch 'upstream/tunnel' into feature/BookB-Pset_GeoScienceModelCommon

larswik committed:
Changes according to latest decisions.

larswik committed:
Update Documentation.md

"eg" => "e.g."

larswik commented:
Current definitions:
| Property | Datatype | Description |
|----------|----------|-------------|
|ModelUseCase|IfcText|Specifies the use case for which the model was created. ModelUseCase could be e.g.: stability assessment, quantity assessment, tendering etc.|
|CreationDate|IfcDate|Creation date of the model.|
|ReleaseDate|IfcDate|Date of release of the model.|
|UpdateDate|IfcDate|Date of update of the model.|
|VersionNumber|IfcLabel|Version tag of the model regarding its descriptive informations.|
|ModelPurpose|IfcLabel|Purpose for which the model was done. Purpose could be e.g.: scientific study, mining, oil reservoir, geotechnics, hydrogeology, simulation or a URI from the codelist https://data.geoscience.earth/ncl/ModelPurpose.|
|ModelStatus|IfcLabel|Status of the model. Status could be e.g.: ended, in-progress, in-development or a URI from the codelist https://data.geoscience.earth/ncl/ModelStatus.|
|Authors|List IfcLabel|Name of the author(s) involved in producing the model.|
|Organizations|List IfcLabel |Name of the organization(s) involved in producing the model.|
|AreaOfInterest|IfcIdentifier|Reference to geographical gazeetteer entries (France, Paris basin etc). <br />>NOTE Geonames (https://www.geonames.org/) could be used in case no other is available.|

pjanck reviewed (approved):
lftm
